packages/workers-shared/asset-server-worker/src/index.ts

```diff
@@ -21,13 +20,17 @@ export default {
 		const url = new URL(request.url);
 		const { pathname } = url;
 
-		const isLocalDevContext = new Uint8Array(ASSETS_MANIFEST).at(0) === 1;
-		const assetsManifest = isLocalDevContext
-			? new NoopAssetsManifest()
-			: new AssetsManifest(ASSETS_MANIFEST);
+		const assetsManifest = new AssetsManifest(ASSETS_MANIFEST);
 		const assetEntry = await assetsManifest.get(pathname);
+		if (!assetEntry) {
+			return new Response("Not Found", { status: 404 });
+		}
 
-		const content = await ASSETS_KV_NAMESPACE.get(assetEntry);
+		const isLocalDevContext = new Uint8Array(ASSETS_MANIFEST).at(0) === 1;
+		const assetKey = isLocalDevContext
+			? assetEntry.path
+			: assetEntry.contentHash;
+		const content = await ASSETS_KV_NAMESPACE.get(assetKey);
```

Comment:
We should avoid any dev specific code in this Worker.
It should just do the same as in production.

Reply:
Not sure that's possible (unless I'm missing something obvious!)

- We need to pass into `ASSETS_KV_NAMESPACE.get()` the pathname in dev, but the content hash in prod.
- We also want to search the manifest in the same way in prod and dev, so therefore the manifest should be the same shape in prod and dev
- In the production manifest, each content hash gets 16 bytes

If we want the ASW to follow exactly the same code in prod and dev, I think we would have to reversibly encode any filepaths into those 16 bytes. I'm not sure that's possible, so unfortunately I do think we need to have a minor divergence here... :(

Reply:
Urgh!!! So we only have 16 bytes to play with... 😢  That is very sad.